packages/dynamic_layouts/lib/src/wrap_layout.dart

```diff
@@ -0,0 +1,270 @@
+///  * [SliverGridLayout], which represents an arbitrary tile layout.
+///  * [DynamicSliverGridGeometry], which represents the size and position of a
+///     single tile in a grid.
+///  * [SliverGridDelegate.getLayout], which returns this object to describe the
+///    delegate's layout.
+///  * RenderDynamicSliverGrid], which uses this class during its
+///    RenderDynamicSliverGrid.performLayout] method.
+class SliverGridWrappingTileLayout extends DynamicSliverGridLayout {
+  /// Creates a layout that uses dynamic sized and spaced tiles.
+  ///
+  /// All of the arguments must not be null and must not be negative.
+  SliverGridWrappingTileLayout({
+    required this.mainAxisSpacing,
+    required this.crossAxisSpacing,
+    required this.childMainAxisExtent,
+    required this.childCrossAxisExtent,
+    required this.crossAxisExtent,
+    required this.scrollDirection,
+  })  : assert(mainAxisSpacing != null && mainAxisSpacing >= 0),
+        assert(crossAxisSpacing != null && crossAxisSpacing >= 0),
+        assert(childMainAxisExtent != null && childMainAxisExtent >= 0),
+        assert(childCrossAxisExtent != null && childCrossAxisExtent >= 0),
+        assert(crossAxisExtent != null && crossAxisExtent >= 0),
+        assert(scrollDirection != null &&
+            (scrollDirection == Axis.horizontal ||
+                scrollDirection == Axis.vertical));
+
+  /// The direction in wich the layout should be built.
+  final Axis scrollDirection;
+
+  /// The extent of the child in the non-scrolling axis.
+  final double crossAxisExtent;
+
+  /// The number of logical pixels between each child along the main axis.
+  final double mainAxisSpacing;
+
+  /// The number of logical pixels between each child along the cross axis.
+  final double crossAxisSpacing;
+
+  /// The number of pixels from the leading edge of one tile to the trailing
+  /// edge of the same tile in the main axis.
+  final double childMainAxisExtent;
+
+  /// The number of pixels from the leading edge of one tile to the trailing
+  /// edge of the same tile in the cross axis.
+  final double childCrossAxisExtent;
+
+  /// The model that is used internally to keep track of how much space is left
+  /// and how much has been used.
+  final List<_RunMetrics> model = <_RunMetrics>[
+    _RunMetrics(maxSliver: 0.0, currentSizeUsed: 0.0, scrollOffset: 0.0)
+  ];
+
+  // In this case the function getGeometryForChildIndex only works to get
+  // the geometry for the child, where the layout actually happens is in
+  // updateGeometryForChildIndex.
+  @override
+  DynamicSliverGridGeometry getGeometryForChildIndex(int index) {
+    return DynamicSliverGridGeometry(
+      scrollOffset: 0,
+      crossAxisOffset: 0,
+      mainAxisExtent: childMainAxisExtent,
+      crossAxisExtent: childCrossAxisExtent,
+    );
+  }
+
+  @override
+  DynamicSliverGridGeometry updateGeometryForChildIndex(
+      int index, Size childSize) {
```

Comment:
nit: get nicer formatting out of the formatter by adding a trailing comma after `childSize`.